lnwallet/commitment.go

```diff
@@ -192,16 +192,35 @@ type ScriptInfo struct {
 
 // CommitScriptToSelf constructs the public key script for the output on the
 // commitment transaction paying to the "owner" of said commitment transaction.
-// If the other party learns of the preimage to the revocation hash, then they
-// can claim all the settled funds in the channel, plus the unsettled funds.
-func CommitScriptToSelf(selfKey, revokeKey *btcec.PublicKey, csvDelay uint32) (
+// The `initiator` argument should correspond to the owner of the commitment
+// tranasction which we are generating the to_local script for. If the other
+// party learns of the preimage to the revocation hash, then they can claim all
+// the settled funds in the channel, plus the unsettled funds.
+func CommitScriptToSelf(chanType channeldb.ChannelType, initiator bool,
+	selfKey, revokeKey *btcec.PublicKey, csvDelay, leaseExpiry uint32) (
 	*ScriptInfo, error) {
 
-	toLocalRedeemScript, err := input.CommitScriptToSelf(
-		csvDelay, selfKey, revokeKey,
-	)
-	if err != nil {
-		return nil, err
+	var toLocalRedeemScript []byte
+	switch {
+	// If we are the initiator of a leased channel, then we have an
+	// additional CLTV requirement in addition to the usual CSV requirement.
+	case initiator && chanType.HasLeaseExpiration():
+		var err error
+		toLocalRedeemScript, err = input.LeaseCommitScriptToSelf(
+			selfKey, revokeKey, csvDelay, leaseExpiry,
```

Comment:
Re the above suggestion, I think it makes sense given that `leaseExpiry` is only ever used if the channel is a leased one.